index.html

```diff
@@ -884,7 +884,30 @@ <h2>The <code>application/credential-claims-set-1.1+json</code> Media Type</h2>
   </table>
   <div class="issue" data-number="42"></div>
 </section>
+</section>
 
+<section id="appendix">
+  <h2>Appendix</h2>
+  <section>
+    <h3>Example VC Data Model Mapping</h3>
+    <p>This is one possible unidirectional mapping between 2.0 VC-JWTs and the VC Data Model; other such mappings are possible.</p>
+    <pre class="example">
+{
+  "@context": [
+    "https://www.w3.org/ns/credentials/v2",
+    {
+        "iss": {"@id": "https://www.w3.org/2018/credentials#issuer"},
+        "sub": {"@id": "https://www.w3.org/2018/credentials#credentialSubject"},
+        "nbf": {"@id": "https://www.w3.org/2018/credentials#issuanceDate"}
+    }
+  ],
+  "iss": "did:example:123",
+  "sub": "did:example:456",
+  "nbf": 1516239022
```

Comment:
This value is non-conformant with the VC Data Model for two reasons: 

1) It uses `issuanceDate` which has been replaced with `validFrom` in the VC 2.0 data model: https://w3c.github.io/vc-data-model/#validity-period , and 
2) It is a non-conformant date time string per the VCDM.

This was among the concerns I highlighted at the face-to-face meeting. 

You can't /just/ use a JSON-LD context to do the transformation (though, that might get you /most/ of the way there). You also need to define some transformation rules that end up looking like [JSLT](https://schibsted.com/blog/jslt-open-source-language-json-processing/).

Reply:
Yes, this is why it can't be "lossless".

I think a program could get closer.

Reply:
@OR13,

> Yes, this is why it can't be "lossless".

As discussed at the F2F, I think having unidirectional formats that are called out as such with demonstrable and fully functional translation rules (as expressed in a spec, by this group or otherwise) is fine. So it's not "lossless" here, IMO, because you never go from the base credential format *to* this other format, only the other way. And, to my knowledge, the base credential format can express what's in `nbf` with an equivalent representation.

What's needed here are the translation steps that a developer of a translator library would follow to convert from `nbf` to a proper `validFrom` value. Again, it's fine that you can't go the other way -- because it's presumed that this format always starts natively as a JWT. It's just vital to demonstrate that someone can use a library (that implements the rules associated with this format) to convert it to the base credential format in the VCDM.

Reply:
Here is a repo demonstrating a "undirectional mapping" https://github.com/transmute-industries/example-mapping-from-jwt-to-jsonld

Its my understanding that the VCWG is not required to define this mapping, and that there is no requirement that there be only one mapping.

Perhaps we should simply close this PR, since it is not necessary?

@dlongley @msporny should we simply close this PR?

Reply:
@OR13,

> Its my understanding that the VCWG is not required to define this mapping, and that there is no requirement that there be only one mapping.

In order for the VCWG to recommend a particular "unidirectional" representation, `X`:

1. A mapping must be defined such that people can demonstrably implement a function that will transform any instance of `X` to the base media type.
2. The mapping has to be defined in a spec that the VCWG can reference.

So, yes, the mapping does not have to be defined in a spec the VCWG produces. Defining it in a VCWG spec is just one option. However, the mapping must be defined somewhere that the VCWG can reference -- and it must be demonstrably functional. Otherwise, the consequence will be that the VCWG cannot recommend `X`.

As for whether to close this PR... that's just a matter of how the above will be executed. It can happen in any number of different ways so it's not really for me to say.

Reply:
For 1, will a github repo suffice?
For 2, does that mean normatively or informatively.

I fear your interpretation leads to a very poor experience for consumers, who never intend to actually "use" the "core data model mapping".

Reply:
@OR13,

For 1, I expect the WG will want some kind of stable spec text -- that's a question for the WG.
For 2, I think the WG might accept a well-defined, stable, but informative reference, I don't know.

> I fear your interpretation leads to a very poor experience for consumers, who never intend to actually "use" the "core data model mapping".

The purpose for providing the mapping is to enable other people to consume the representation as "the core data model". For example, digital wallet providers that want to support common rendering of credentials from differently sourced representations. This is an important aspect of the VC three party model -- the individual parties are independent and don't know who or how other parties will be consuming VCs. More closed models or use cases don't have this attribute nor this need -- and often they also don't even need any new standards to support whatever it is they are doing with their specialized data.

So, the mapping spec creator doesn't have to know who the consumers are, nor do they have to be one of them. They are just supporting others so that their representation can be considered one that encodes verifiable credentials for general consumption. Of course, if no mapping is provided or there's no intention to properly support consumers of the core data model, then there's no reason for the WG to consider such a representation in scope. Such a representation would be it's "own thing" and that's fine -- it just doesn't have any reason to be referenced by a spec in this group.

Reply:
@dlongley it sounds like the mapping would best be left to the wallet software implementers... It sounds like an implementation detail, not something for this working group to define.

I am increasingly leaning towards considering the day 3 resolution unworkable.

You seem to be asking working group members to do work which they do not see as valuable, and which would be better done, by wallet software implementers.

Reply:
@OR13,

>  ...it sounds like the mapping would best be left to the wallet software implementers...

Wallet implementers will need to implement the mapping, however, each should definitely not be defining their own artisanal mappings as that completely misses the point of standardization and interop. Wallet implementers need to be able to come find the spec associated with the representation that defines a mapping and implement it. If a representation is only ever to be consumed by a well-known, small group of people -- and none of them are consuming the core data model -- then there's no reason for it to have anything to do with this WG.

In fact, often people in those kinds of situations don't need any new standards because there are already existing standards for securing data that is modeled in arbitrary ways.

> You seem to be asking working group members to do work which they do not see as valuable, and which would be better done, by wallet software implementers.

A representation that lacks a mapping spec for software developers (such as wallet providers) to implement is out of scope for the VCWG. Again, it's fine if the people defining some representation (of something) don't care to support consumers of the core data model, but then that representation is simply out of scope. It's just not the right place to do that kind of work here. The work being done here is to support those consumers.

cc: @jandrieu you may have some thoughts to contribute here.